### PR TITLE
Extent the range of `geom_abline()` outside the panel range

### LINKED ISSUE #4024: Can geom_abline() draw lines a bit longer than the actual Coord range?
Someone told me `geom_abline()` doesn't look nice when we increase the `size`. To be fair, `geom_function()` seems better at the first example, but it has the same problem when the range of the scale is shorter or equal to the coord's range. I think this is intentional, but it might be nice if there's some option to draw lines to outside of the plot area.

(Btw, maybe this is a good chance to replace `GeomAbline` with `GeomFunction`...?)

``` r
library(ggplot2)
library(tibble)
library(patchwork)

p <- ggplot(tibble(x = c(-5, 5), y = c(-4, 6))) +
  geom_line(aes(x, y), size = 10, color = "#cc0000") +
  coord_fixed(xlim = c(-1, 1), ylim = c(-1, 1), expand = FALSE) +
  theme_bw()

p1 <- p +
  geom_abline(slope = 1, intercept = 1, size = 10, color = "#000088", alpha = 0.7) +
  ggtitle("abline")

p2 <- p +
  geom_function(fun = ~ .x + 1, size = 10, color = "#000088", alpha = 0.7) +
  ggtitle("function")

p1 * p2
```

<sup>Created on 2020-05-24 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

``` r
library(ggplot2)

ggplot() +
  geom_function(fun = ~ .x + 1, size = 10, color = "#000088", alpha = 0.7) +
  xlim(-1, 1) +
  ggtitle("function") + 
  coord_fixed(xlim = c(-1, 1), ylim = c(-1, 1), expand = FALSE) +
  theme_bw()
```

<sup>Created on 2020-05-24 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### PR DESCRIPTION
**Pull request by thomasp85**
Fix #4024

This PR only fixes the issue in `geom_abline()` leaving `geom_function()` as it is because it is not really feasible to fix it right now/for this

### CODE REVIEW

File: R/geom-abline.r [hunk 1]
```diff
@@ -127,6 +127,12 @@ GeomAbline <- ggproto("GeomAbline", Geom,
   draw_panel = function(data, panel_params, coord) {
     ranges <- coord$backtransform_range(panel_params)
 
+    if (T || coord$clip == "on" && coord$is_linear()) {
```
Comment (yutannihilation) [inline R/geom-abline.r:130]: What's the intention of this `T`...?

Comment (thomasp85) [reply R/geom-abline.r:130]: 😵 Thank god for you reviewing it

Comment (yutannihilation) [reply R/geom-abline.r]: Ah, for debugging. I see.

**Comment by yutannihilation** [comment]
Looks good. Could you update visual cases?

**Comment by thomasp85** [comment]
yes, of course :-)

**Review by yutannihilation — APPROVED** [review]

**Pull request merged**